### PR TITLE
Reduce bundle size

### PR DESCRIPTION
**Pull request by brenelz**
- [X] I have read the [Contribution guide](https://github.com/solidjs/solid-docs/blob/main/CONTRIBUTING.md)
- [X] This PR references an issue (except for typos, broken links, or other minor problems)

### Description(required)
Playing around a bit with bundle size and lighthouse scores on the docs. I wondered if I could improve things by keeping our syntax highligher(shiki) on the server.  I will do a bit of testing to see if it actually improves things.

Im realizing its hard to test because of the netlify deploy preview drawer. 
(Found a link without the drawer https://66f86aa87b3ff50008d7d297--solid-docs.netlify.app/)

## Comparison
Before
image
image

After
image
image

### CODE REVIEW

**Comment by brenelz** [comment]
@atilafassina do you have any thoughts on this? Is it a good approach to reduce bundle size or do we want the highlighter on the client?

**Comment by atilafassina** [comment]
> @atilafassina do you have any thoughts on this? Is it a good approach to reduce bundle size or do we want the highlighter on the client?

@brenelz 
the highlighter on the client was a quick-and-dirty thing that was hacked around for the PoC stage which I never had time to get back to. So I totally appreciate the time and effort you're putting into this!!

**Comment by brenelz** [comment]
Just re commenting here for visibility. Is this worth investing some time in? I think its mostly there

**Comment by LadyBluenotes** [comment]
> Just re commenting here for visibility. Is this worth investing some time in? I think its mostly there

I'd say this is worth investing time in, especially if you have an idea of how to do it and it isn't too complicated

File: src/ui/layout/hero-code-snippet.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
+
+import { cache, createAsync } from "@solidjs/router";
```
Comment (atilafassina) [inline src/ui/layout/hero-code-snippet.tsx:2]:
```suggestion
import { query, createAsync } from "@solidjs/router";
```

File: src/ui/layout/hero-code-snippet.tsx [hunk 2]
```diff
@@ -17,16 +18,17 @@ function Counter() {
 
 export const snippetLines = counterTxt.split("\n");
 
-const renderCode = async () => {
+const renderCode = cache(async () => {
```
Comment (atilafassina) [inline src/ui/layout/hero-code-snippet.tsx:21]:
```suggestion
const renderCode = query(async () => {
```

**Review by atilafassina — CHANGES_REQUESTED** [review]
small nit! 🏆

File: src/ui/layout/hero-code-snippet.tsx (hunk 1 above)
Comment (brenelz) [reply src/ui/layout/hero-code-snippet.tsx:2]: Looks like we are on an older version of the router. Should i upgrade the router or leave it as is?

**Review by atilafassina — APPROVED** [review]

File: src/ui/layout/hero-code-snippet.tsx (hunk 1 above)
Comment (atilafassina) [reply src/ui/layout/hero-code-snippet.tsx:2]: oh really?! I thought that was updated already.. nevermind then

**Pull request merged**